drivers/serial/uart_stm32.c

```diff
@@ -643,6 +1119,14 @@ static const struct uart_driver_api uart_stm32_driver_api = {
 	.irq_update = uart_stm32_irq_update,
 	.irq_callback_set = uart_stm32_irq_callback_set,
 #endif	/* CONFIG_UART_INTERRUPT_DRIVEN */
+#ifdef CONFIG_UART_ASYNC_API
+	.callback_set = uart_stm32_callback_set,
+	.tx = uart_stm32_tx,
+	.tx_abort = uart_stm32_tx_abort,
+	.rx_enable = uart_stm32_rx_enable,
+	.rx_disable = uart_stm32_rx_disable,
+	.rx_buf_rsp = uart_stm32_rx_buf_rsp,
```

Comment:
We might want to make clear these functions are async/dma dedicated.
So maybe `uart_stm32_tx` should be renamed to `uart_stm32_async_tx` and so on ....

Reply:
will change to reflect that.